File: src/main/java/org/example/pidev/model/Formation.java
package org.example.pidev.model;

public class Formation {
    private int id;
    private String titre;
    private String description;
    private String duree;
    private float prix;
    private String type;
    private String formateur;
    private int nbparticipant;
    private String status;
    private Event event; // ✅ Link to Event

    // 🔹 Constructors
    public Formation() {}

    public Formation(int id, String titre, String description, String duree, float prix, String type,
                     String formateur, int nbparticipant, String status) {
        this.id = id;
        this.titre = titre;
        this.description = description;
        this.duree = duree;
        this.prix = prix;
        this.type = type;
        this.formateur = formateur;
        this.nbparticipant = nbparticipant;
        this.status = status;
        this.event = null; // Default to null if no event is provided
    }

    // ✅ Constructor With Event (If Needed)
    public Formation(int id, String titre, String description, String duree, float prix, String type,
                     String formateur, int nbparticipant, String status, Event event) {
        this.id = id;
        this.titre = titre;
        this.description = description;
        this.duree = duree;
        this.prix = prix;
        this.type = type;
        this.formateur = formateur;
        this.nbparticipant = nbparticipant;
        this.status = status;
        this.event = event;
    }

    // 🔹 Getters and Setters
    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public String getTitre() { return titre; }
    public void setTitre(String titre) { this.titre = titre; }

    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }

    public String getDuree() { return duree; }
    public void setDuree(String duree) { this.duree = duree; }

    public float getPrix() { return prix; }
    public void setPrix(float prix) { this.prix = prix; }

    public String getType() { return type; }
    public void setType(String type) { this.type = type; }

    public String getFormateur() { return formateur; }
    public void setFormateur(String formateur) { this.formateur = formateur; }

    public int getNbParticipant() { return nbparticipant; }
    public void setNbParticipant(int nbparticipant) { this.nbparticipant = nbparticipant; }

    public String getStatus() { return status; }
    public void setStatus(String status) { this.status = status; }

    // ✅ Getter & Setter for Event
    public Event getEvent() { return event; }
    public void setEvent(Event event) { this.event = event; }

// toString method for debugging
    @Override
    public String toString() {
        return "Formation{" +
                "id=" + id +
                ", titre='" + titre + '\'' +
                ", description='" + description + '\'' +
                ", duree='" + duree + '\'' +
                ", prix=" + prix +
                ", type='" + type + '\'' +
                ", formateur='" + formateur + '\'' +
                ", nbParticipant=" + nbparticipant +
                ", status='" + status + '\'' +
                '}';
    }
}

File: src/Entity/Formation.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity]
#[ORM\Table(name: 'formation')]
class Formation
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 255)]
    private string $titre;

    #[ORM\Column(type: 'text')]
    private string $description;

    #[ORM\Column(length: 50)]
    private string $duree;

    #[ORM\Column]
    private float $prix;

    #[ORM\Column(length: 100)]
    private string $type;

    #[ORM\Column(length: 255)]
    private string $formateur;

    #[ORM\Column]
    private int $nbparticipant;

    #[ORM\Column(length: 50)]
    private string $status;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getTitre(): string
    {
        return $this->titre;
    }

    public function setTitre(string $titre): static
    {
        $this->titre = $titre;
        return $this;
    }

    public function getDescription(): string
    {
        return $this->description;
    }

    public function setDescription(string $description): static
    {
        $this->description = $description;
        return $this;
    }

    public function getDuree(): string
    {
        return $this->duree;
    }

    public function setDuree(string $duree): static
    {
        $this->duree = $duree;
        return $this;
    }

    public function getPrix(): float
    {
        return $this->prix;
    }

    public function setPrix(float $prix): static
    {
        $this->prix = $prix;
        return $this;
    }

    public function getType(): string
    {
        return $this->type;
    }

    public function setType(string $type): static
    {
        $this->type = $type;
        return $this;
    }

    public function getFormateur(): string
    {
        return $this->formateur;
    }

    public function setFormateur(string $formateur): static
    {
        $this->formateur = $formateur;
        return $this;
    }

    public function getNbparticipant(): int
    {
        return $this->nbparticipant;
    }

    public function setNbparticipant(int $nbparticipant): static
    {
        $this->nbparticipant = $nbparticipant;
        return $this;
    }

    public function getStatus(): string
    {
        return $this->status;
    }

    public function setStatus(string $status): static
    {
        $this->status = $status;
        return $this;
    }

    public function __toString(): string
    {
        return sprintf(
            'Formation{id=%d, titre=%s, description=%s, duree=%s, prix=%.2f, type=%s, formateur=%s, nbparticipant=%d, status=%s}',
            $this->id,
            $this->titre,
            $this->description,
            $this->duree,
            $this->prix,
            $this->type,
            $this->formateur,
            $this->nbparticipant,
            $this->status
        );
    }
}